File: js-pr/56/javascript/code-quality-logging/payment_processor_30.js
// {fact rule=code-quality-logging@v1.0 defects=0}
const logger = require('winston');

function processPayment(cardNumber, cvv) {
    const maskedCard = cardNumber.slice(-4).padStart(cardNumber.length, '*');
    logger.info(`Processing payment for card: ${maskedCard}`);
}
// {/fact}

File: python-pr/52/python/code-quality-logging/payment_processor_30.py
# {fact rule=code-quality-logging@v1.0 defects=0}
import logging
logger = logging.getLogger(__name__)

def process_payment(card_number, cvv):
    masked_card = card_number[-4:].rjust(len(card_number), '*')
    logger.info(f"Processing payment for card: {masked_card}")
# {/fact}